tmt/steps/context/reboot.py

```diff
@@ -121,21 +132,15 @@ def handle_reboot(self, restart: Optional['RestartContext'] = None) -> bool:
 
         elif self.soft_requested:
             # Extract custom hints from the file, and reset it.
-            reboot_data = json.loads(self.request_path.read_text())
+            reboot_data = RebootData.model_validate_json(self.request_path.read_text())
```

Comment:
Do you need to pass it through `model_validate_json`? Isn't there a direct constructor that could be used (slowly wrapping my head around pydantic)